apps/remote_control/lib/lexical/remote_control/analyzer/requires.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Lexical.RemoteControl.Analyzer.Requires do
+  alias Lexical.Ast.Analysis
+  alias Lexical.Ast.Analysis.Require
+  alias Lexical.Ast.Analysis.Scope
+  alias Lexical.Document.Position
+
+  def at(%Analysis{} = analysis, %Position{} = position) do
+    case Analysis.scopes_at(analysis, position) do
+      [%Scope{} = scope | _] ->
+        scope.requires
+        |> Enum.filter(fn %Require{} = require ->
+          require_end = require.range.end
+
+          if require_end.line == position.line do
+            require_end.character <= position.character
+          else
+            require_end.line < position.line
+          end
```

Comment:
We have this logic duplicated or *nearly* duplicated in a lot of places, but sometimes with superficial differences that can make it hard to know whether the code is equivalent. For instance, this code here has `require_end OP position` whereas the similar code in `Uses` has `position OP use_end`.

Can we define a `Position.compare/2` and start using it instead?

```suggestion
          Position.compare(position, require.range.end) in [:eq, :gt]
```

This could be in a separate PR that could audit the codebase for other places where code like this could be rewritten in terms of `Position.compare/2`.

Reply:
This is a good call,  was considering this very suggestion, but felt it was out of scope for the PR.